pkg/customresource/repositoryserver.yaml

```diff
@@ -0,0 +1,278 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: repositoryservers.cr.kanister.io
+spec:
+  group: cr.kanister.io
+  names:
+    kind: RepositoryServer
+    listKind: RepositoryServerList
+    plural: repositoryservers
+    singular: repositoryserver
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: RepositoryServer manages the lifecycle of Kopia Repository Server
+          within a Pod
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: RepositoryServerSpec is the specification for the RepositoryServer
+            properties:
+              repository:
+                description: Repository details for the purpose of establishing a
+                  connection
+                properties:
+                  hostname:
+                    type: string
+                  passwordSecretRef:
+                    description: SecretReference represents a Secret Reference. It
+                      has enough information to retrieve secret in any namespace
```

Comment:
All the reference descriptions seem generic. Should we change all of them to describe specific secrets?